internal/runner/operation_project_destroy.go

```diff
@@ -0,0 +1,71 @@
+package runner
+
+import (
+	"context"
+	"github.com/hashicorp/waypoint/internal/core"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+func (r *Runner) executeDestroyProjectOp(
+	ctx context.Context,
+	job *pb.Job,
+	project *core.Project,
+) (*pb.Job_Result, error) {
+	client := project.Client()
+	destroyProjectOp, ok := job.Operation.(*pb.Job_DestroyProject)
+	if !ok {
+		// this shouldn't happen since the call to this function is gated
+		// on the above type match.
+		panic("operation not expected type")
+	}
+
+	// Update the project state to indicate that it's being destroyed
+	// TODO: Do we need to do something with the response here?
```

Comment:
Probably not, UpsertProject will return the same Project proto message that was within the request i.e. the value you set within `&pb.UpsertProjectRequest{Project: ... }`

https://github.com/hashicorp/waypoint/blob/d0b40daa74d483a3a8838e43f93bfada5736270d/pkg/server/singleprocess/service_project.go#L20-L42